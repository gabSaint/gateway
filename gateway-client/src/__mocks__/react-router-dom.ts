import React, { FunctionComponent } from "react";

const useParams = jest.fn();
const useHistory = jest.fn();

const Link: FunctionComponent = ({ children }) =>
  React.createElement("div", null, children);

export { useParams, useHistory, Link };
