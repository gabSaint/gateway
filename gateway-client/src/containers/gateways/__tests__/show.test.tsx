import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import renderer from "react-test-renderer";
import { useParams } from "react-router-dom";

import ShowGateway from "../show";

describe("ShowGateway", () => {
  let wrapperShowGateway: ShallowWrapper | ReactWrapper;

  beforeEach(() => {
    // @ts-ignore
    useParams.mockImplementationOnce(() => ({ id: "1" }));

    wrapperShowGateway = shallow(<ShowGateway />);
  });

  it("renders correctly", () => {
    // @ts-ignore
    useParams.mockImplementationOnce(() => ({ id: "1" }));

    const tree = renderer.create(<ShowGateway />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Header & Footer", () => {
    it("renders title", () => {
      const title = wrapperShowGateway.find("h2");
      expect(title).toHaveLength(1);
    });

    it("renders create new peripheral button", () => {
      const newPeripheralButton = wrapperShowGateway.find(
        "[data-test='button-add-peripheral']"
      );
      expect(newPeripheralButton).toHaveLength(1);
    });
  });
});
