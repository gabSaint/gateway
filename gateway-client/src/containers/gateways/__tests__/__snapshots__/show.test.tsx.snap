// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowGateway renders correctly 1`] = `
Array [
  <h2>
    <div>
      Gateways
    </div>
     - 
  </h2>,
  <div>
    <button
      data-test="button-add-peripheral"
    >
      Add peripheral
    </button>
  </div>,
]
`;
